rules/scss.js

```diff
@@ -60,4 +63,11 @@ module.exports = {
   'scss/partial-no-import': true,
   // Disallow redundant nesting selectors (`&`).
   'scss/selector-no-redundant-nesting-selector': true,
+  // Disallow dollar variables within a stylesheet.
+  'scss/no-dollar-variables': null,
+  // Disallow duplicate dollar variables within a stylesheet.
+  'scss/no-duplicate-dollar-variables': [
+    true,
+    {ignoreInsideAtRules: ['if', 'mixin']},
```

Comment:
Let's add `function` to that list, I ran into a lot of errors related to that when linting polaris-styleguide.

Reply:
@kaelig just tested this on web while tophattin' https://github.com/Shopify/stylelint-config-shopify/pull/42/files. Couple cases of `include` came up as failures, should we add it to the list?

Reply:
This rule doesn't seem to make any attempt to actually understand how sass does variable scoping. IF we're adding `if` to this then we also need an `else` as the following results in an linting error:

```scss
@function foo() {
  @if true {
    $foo: 200;
  } @else {
    $bar: 400;
  }
}

@function bar() {
  @if true {
    $foo: 200;
  } @else {
    $bar: 900;
  }
}

```

Rather than playing a game of whach-a-mole with every at-directive we can come up with (`else`, `include`, etc) I think it would be better to disable this rule entirely.

Reply:
Yah, i'm starting to think that as well. It has been a bit of wack-a-mole. Let's wait to hear from Kaelig before disabling.